webui/src/App.js

```diff
@@ -21,86 +21,168 @@ import {
   Link
 } from "react-router-dom";
 
-// Amount of panels.
-const num_panels = 4;
+// Maximum number of historical sensor values to retain
+const max_size = 1000;
 
-// Keep track of the current thresholds fetched from the backend.
-// Make it global since it's used by many components.
-let kCurThresholds = new Array(num_panels).fill(0);
+// Reference to history of current sensor values.
+// The values are stored in a mutable array in a ref so that they are not
+// subject to the React render cycle, for performance reasons.
+// The Context is to make them easy to access from nested components.
+const CurValuesRefContext = React.createContext();
```

Comment:
Similarly, I think this can be CUR_VALUES_CONTEXT (or similar).

Reply:
I think MAX_SIZE sounds good, but CUR_VALUES_CONTEXT I'm not so sure about.

From the google style guide you linked, 

> Before choosing constant case, consider whether the field really feels like a deeply immutable constant. For example, if any of that instance's observable state can change, it is almost certainly not a constant. Merely intending to never mutate the object is generally not enough.

This one doesn't feel like a constant to me, since the value of `useContext(CUR_VALUES_CONTEXT)` depends on the value passed to the nearest containing `CUR_VALUES_CONTEXT.Provider`. Also it contains a ref with mutable state.

Reply:
My thought about putting it in a context was, since I was getting rid of some global variables (technically module-level), it would still be easy to access the values/thresholds with a context without passing the props through every layer. Looking at it now though, there aren't that many layers, so I'm thinking I'll remove it anyway. I can put it back if you want to keep it.

Reply:
Yeah I think I agree with you with naming.

Props I think is fine.